File: examples/app-yaml-config/sdk/mjs/12-resolver-security.mjs
#!/usr/bin/env node
// F03 / story 04 — security guard rejects __proto__ etc.
import { createResolver, SecurityError } from "@ployglot/runtime-template-resolver";

const r = createResolver();
const out = {};

try {
  await r.resolve("{{__proto__.polluted}}", { __proto__: { polluted: "evil" } });
  out.proto_attempt = "<unexpected: did not throw>";
} catch (e) {
  out.proto_attempt = e instanceof SecurityError
    ? "<caught: SecurityError>"
    : `<caught: ${e.constructor.name}>`;
}

try {
  await r.resolve("{{constructor.name}}", {});
  out.constructor_attempt = "<unexpected: did not throw>";
} catch (e) {
  out.constructor_attempt = e instanceof SecurityError
    ? "<caught: SecurityError>"
    : `<caught: ${e.constructor.name}>`;
}

console.log(JSON.stringify(out, null, 2));

File: examples/app-yaml-config/sdk/py/12-resolver-security.py
#!/usr/bin/env python3
"""F03 / story 04 — security guard rejects __class__/__dict__ etc."""
from __future__ import annotations

import asyncio
import json

from runtime_template_resolver import SecurityError, create_resolver


async def main():
    r = create_resolver()
    out = {}

    try:
        await r.resolve("{{__class__.__name__}}", {})
        out["class_attempt"] = "<unexpected: did not raise>"
    except SecurityError:
        out["class_attempt"] = "<caught: SecurityError>"
    except Exception as e:
        out["class_attempt"] = f"<caught: {type(e).__name__}>"

    try:
        await r.resolve("{{__dict__.foo}}", {})
        out["dict_attempt"] = "<unexpected: did not raise>"
    except SecurityError:
        out["dict_attempt"] = "<caught: SecurityError>"
    except Exception as e:
        out["dict_attempt"] = f"<caught: {type(e).__name__}>"

    print(json.dumps(out, indent=2))


if __name__ == "__main__":
    asyncio.run(main())
